jenkins/deployment/community/cortx-community-deploy.groovy

```diff
@@ -15,156 +14,170 @@ pipeline {
+                    popd
+            '''
+            }
+        }
+        stage('Execute cortx build script') {
             steps {
                 script { build_stage = env.STAGE_NAME }
-                sh label: 'executing cortx build image script', script: '''
+                sh label: 'Executing cortx build image script on Primary node', script: '''
                 pushd solutions/community-deploy/cloud/AWS
                     export CORTX_RE_BRANCH=${CORTX_RE_BRANCH}
-                    export CORTX_TAG=${CORTX_TAG}
-                    sleep 120 
-                    ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${AWS_IP}" "export CORTX_RE_BRANCH=$CORTX_RE_BRANCH; git clone $CORTX_RE_REPO -b $CORTX_RE_BRANCH; pushd /home/centos/cortx-re/solutions/community-deploy; time sudo ./build-cortx.sh -b ${CORTX_TAG}"
-                popd
+                    export PRIMARY_PUBLIC_IP=$(cat ip_public.txt | jq '.[0]'| tr -d '",[]')
+                    ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${PRIMARY_PUBLIC_IP}" "export CORTX_RE_BRANCH=$CORTX_RE_BRANCH; git clone $CORTX_RE_REPO -b $CORTX_RE_BRANCH; pushd /home/centos/cortx-re/solutions/community-deploy; time sudo ./build-cortx.sh -b ${CORTX_TAG}"
+                    popd
             '''
             }
         }
-
-        stage ('EC2 connection prerequisites') {
+        stage('Setup K8s cluster') {
             steps {
                 script { build_stage = env.STAGE_NAME }
-                sh label: 'Changing root password and creating hosts file', script: '''
+                sh label: 'Setting up K8s cluster on EC2', script: '''
                 pushd solutions/community-deploy/cloud/AWS
-                    export ROOT_PASSWORD=${ROOT_PASSWORD}              
-                    ssh -i cortx.pem -o StrictHostKeyChecking=no centos@"${AWS_IP}" "export ROOT_PASSWORD=$ROOT_PASSWORD && echo $ROOT_PASSWORD | sudo passwd --stdin root && pushd /home/centos/cortx-re/solutions/kubernetes && echo "'"hostname=$HOSTNAME,user=root,pass="'" > hosts && sed -i 's,pass=.*,pass=$ROOT_PASSWORD,g' hosts && cat hosts"
+                    export PRIMARY_PUBLIC_IP=$(cat ip_public.txt | jq '.[0]'| tr -d '",[]')
+                    export WORKER_IP=$(cat ip_public.txt | jq '.[0]','.[1]','.[2]','.[3]' | tr -d '",[]')
+                    export HOST1=$(cat ec2_hostname.txt | jq '.[0]'| tr -d '",[]')
+                    ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${PRIMARY_PUBLIC_IP}" "sudo -- sh -c 'pushd /home/centos/cortx-re/solutions/kubernetes && ./cluster-setup.sh true && sed -i 's,cortx-docker.colo.seagate.com,${HOST1}:8080,g' /etc/docker/daemon.json && systemctl restart docker && sleep 120'"
+                    for wp in $WORKER_IP;do ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@${wp} "sudo -- sh -c 'sed -i 's,cortx-docker.colo.seagate.com,${HOST1}:8080,g' /etc/docker/daemon.json && systemctl restart docker && sleep 120'";done
```

Comment:
Please add sleep after modifying the docker restart as it takes some time to restart and make the K8s cluster back to normal. Refer - https://github.com/Seagate/cortx-re/blob/main/jenkins/deployment/community/cortx-community-deploy.groovy#L139

Reply:
sleep command is already invoked in stage